projects/kit/components/input-year/input-year.directive.ts

```diff
@@ -0,0 +1,97 @@
+import {computed, Directive, effect, inject, Input, signal} from '@angular/core';
+import {MaskitoDirective} from '@maskito/angular';
+import {maskitoNumberOptionsGenerator} from '@maskito/kit';
+import {tuiAsControl, TuiControl, tuiValueTransformerFrom} from '@taiga-ui/cdk/classes';
+import {TUI_ALLOW_SIGNAL_WRITES} from '@taiga-ui/cdk/constants';
+import {TuiCalendarYear} from '@taiga-ui/core/components/calendar';
+import {
+    tuiInjectAuxiliary,
+    tuiTextfieldIconBinding,
+    TuiWithTextfield,
+} from '@taiga-ui/core/components/textfield';
+import {tuiDropdownOpen} from '@taiga-ui/core/directives/dropdown';
+import {TuiItemsHandlersValidator} from '@taiga-ui/core/directives/items-handlers';
+import {tuiMaskito} from '@taiga-ui/kit/utils';
+
+import {TUI_INPUT_YEAR_OPTIONS} from './input-year.options';
+
+@Directive({
+    standalone: true,
+    selector: 'input[tuiInputYear]',
+    providers: [
+        tuiAsControl(TuiInputYearDirective),
+        tuiValueTransformerFrom(TUI_INPUT_YEAR_OPTIONS),
+    ],
+    hostDirectives: [TuiWithTextfield, MaskitoDirective, TuiItemsHandlersValidator],
+    host: {
+        maxlength: '4',
+        inputmode: 'numeric',
+        '[value]': 'value()',
+        '[disabled]': 'disabled()',
+        '(click)': 'toggle()',
+        '(input)': 'onChange(toNumber($event.target.value))',
```

Comment:
```suggestion
        '(input)': 'onChange($event.target.valueAsNumber)',
```

https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/valueAsNumber

Reply:
fixed